src/arrayMethodSort.js

```diff
@@ -3,9 +3,22 @@
 /**
  * Implement method Sort
  */
+
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(cb) {
+    for (let i = 0; i < this.length - 1; i++) {
+      const currElem = this[i];
+      const nextElem = this[i + 1];
+
+      if (cb ? cb(nextElem, currElem, i, this) < 0
+        : nextElem.toString()[0] < currElem.toString()[0]) {
```

Comment:
The comparison of the first character of the toString method can lead to incorrect sorting for numbers greater than 9. Consider using a different default comparison function.